httpx/_multipart.py

```diff
@@ -120,9 +136,9 @@ def render_headers(self) -> bytes:
             if self.filename:
                 filename = format_form_param("filename", self.filename)
                 parts.extend([b"; ", filename])
-            if self.content_type is not None:
-                content_type = self.content_type.encode()
-                parts.extend([b"\r\nContent-Type: ", content_type])
+            for header_name, header_value in self.headers.items():
+                key, val = f"\r\n{header_name}: ".encode(), header_value.encode()
```

Comment:
check should be case insensitive to header
```suggestion
                 key, val = f"\r\n{header_name.lower()}: ".encode(), header_value.encode()
```